components/toast/toast.jsx

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import UnreadBelowIcon from 'components/widgets/icons/unread_below_icon';
+import CloseIcon from 'components/widgets/icons/close_icon';
+
+function empty() {
+    // do nothing
+}
+
+const someBigNum = 1000;
+
+export default class Toast extends React.PureComponent {
+    static propTypes = {
+        jumpTo: PropTypes.func.isRequired, // required?
+        jumpToMessage: PropTypes.string,
+        jumpFadeOutDelay: PropTypes.number,
+        onDismiss: PropTypes.func,
+        order: PropTypes.number,
+        children: PropTypes.element,
+        show: PropTypes.bool.isRequired,
+        extraClasses: PropTypes.string,
+        showOnlyOnce: PropTypes.bool.isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+
+            //once hide is set to true, only mounting again from the caller will show it again. This way, once we have used one button the toast doesn't show up again.
+            // if it should only be shown once, we set it to the value of show.
+            hide: this.props.showOnlyOnce ? !this.props.show : false,
```

Comment:
Should there maybe be a way to completely remove the extra `Toast` components? As it is, we'll just have an invisible component that's always mounted, but not rendered

Reply:
Once I change the permalink view, I'll try to render two toast tops, and most of the time only one (the new messages unread one).

Reply:
changed, now only one is always rendered and the other two are on demand